app/gui/src/project-view/stores/project/executionContext.ts

```diff
@@ -255,14 +255,23 @@ export class ExecutionContext extends ObservableV2<ExecutionContextNotification>
 
   /** TODO: Add docs */
```

Comment:
The function signature is getting more complex; this would be a good time to add docs